db/version_edit_handler.cc

```diff
@@ -27,12 +27,15 @@ VersionEditHandler::VersionEditHandler(
   assert(version_set_ != nullptr);
 }
 
-Status VersionEditHandler::Iterate(log::Reader& reader, std::string* db_id) {
```

Comment:
I see we already have a parameter passed by non-const reference here, but I'd guess it's one of very few in the RocksDB codebase. Maybe we should prevent that pattern from spreading and pass pointers for consistency?

Reply:
hmm. I checked style guide https://google.github.io/styleguide/cppguide.html#Inputs_and_Outputs. The way I see it is that both `reader` and `log_read_status` are kind of input/output parameters, and they are both non-nullable. Therefore, reference seems to conform to the guide. Let me know if I am missing anything here.

Reply:
Actually, `log_read_status` is output parameter. Will update this.

Reply:
Yeah I checked and noticed that too. Maybe my experience with this is from FB's linter or obsolete -- can't really remember. Anyways, leave it up to you.